cli/azd/pkg/apphost/generate.go

```diff
@@ -1309,6 +1309,24 @@ func (b infraGenerator) evalBindingRef(v string, emitType inputEmitType) (string
 			bindingMappedToMainIngress = true
 		}
 
+		hostNameSuffix := func(external bool) string {
+			var suffix string
+			switch emitType {
+			case inputEmitTypeYaml:
```

Comment:
Here is one of details why I usually prefer not using a function inside a function. It is not clear if `inputEmitTypeYaml` is scoped to the parent function or if it is a package global variable =P (at least by the first look at the PR).

Not a blocker, or anything.  I guess these cases might be handled by the compiler similar to macros, avoiding the call-stack.

Reply:
I am guessing you mean `emitType` (since `inputEmitTypeYaml` is a constant) which is the argument to the outer function we closed over.

I did consider pushing this to top level, but it would mean the call sites got a little longer since we have to thread `emitType` into the call, which is sort of why I prefer this style.

Something else I briefly considered was adding:

```go
func (emitType inputEmitType) hostNameSuffix(external bool) string {
   // ... the implementation is the same as the above ...
}
```

and then at all sites doing: `emitType.hostNameSuffix(binding.External)` but I wasn't sure if that was being too clever for my own good.